internal/transport/transport_test.go

```diff
@@ -844,17 +844,11 @@ func (s) TestGracefulClose(t *testing.T) {
 		wg.Add(1)
```

Comment:
Nit: can you switch the docstring for this code block from "Expect the failure for all the follow-up streams because ct has been closed gracefully." to something like "expect errors trying to create new streams after the client transport has been Gracefully Closed (and is in draining)".